framework/src/mesh/MooseMesh.C

```diff
@@ -1708,7 +1708,7 @@ MooseMesh::init()
     // Set custom partitioner
     if (!_custom_partitioner)
       mooseError("If using partitioner type custom you must explicitly provide a partitioner in your input file!");
-    getMesh().partitioner().reset(_custom_partitioner);
+    getMesh().partitioner().reset(_custom_partitioner.get());
```

Comment:
I'm afraid you might have to do something like:
```
getMesh().partitioner().reset(new Partitioner(*_custom_partitioner));
```
i.e. copy-construct a new object from the shared_ptr for libmesh to own in its unique_ptr.  I don't see another way of transferring ownership.